### PR TITLE
Get preview link 

### PR DESCRIPTION
**Pull request by nerdlet**
## Description

Fixes # Add preview and draft revision URL, adds the option to set frontend base url

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Desktop Screenshots
image

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Review by kilemensi — COMMENTED** [review]
👍🏾 @nerdlet 

Quick review....

1. `htaccess`: Shouldn't be in this PR because it will also affect WPEngine once deployed...
2. `acf`: Again, they don't seem to belong with preview but with rest api changes that @KobbyMmo is working on.... I could be wrong though, not sure how you've divided the work.

But the main review: The preview link needs to be adaptable / configurable to the site since we're using multi-site setup. Basically:

1. If running locally, my preview link **_can be_** locally or whatever configured.
2. If running in prod / WPEngine, my preview link would be:

   1. If site is `/promisetracker`, my link should go to `promisetracker.dev.codeforafrica.org`, and
   2. If site is `/covid-19` (confirm the name), my link should go to `outbreak.africa`. 
   
... and the usual, link and format. I think prettier has a plugin for [PHP](https://github.com/prettier/plugin-php).

**Review by kilemensi — COMMENTED** [review]
👍🏾 @KobbyMmo 

Quick review:

1. Include the front end/base url in the HURUmap options (Can't remember whether it was HURUmap site options or just HURUmap). This way, there is just one place to configure things (may be you have already).
2. Return custom preview URL only if this field is set... if not, return the default WordPress preview URL.

File: wp-content/plugins/hurumap/index.php [hunk 1]
```diff
@@ -375,4 +376,8 @@ function hurumap()
 
 hurumap();
 
+
 endif;
+
+
+
```
Comment (KhadijaMahanga) [inline wp-content/plugins/hurumap/index.php]:
```suggestion
```

File: wp-content/themes/hurumap/acf-json/group_6040a3626fa6e.json [hunk 2]
```diff
@@ -0,0 +1,40 @@
+{
+    "key": "group_6040a3626fa6e",
+    "title": "Fontend Url",
```
Comment (KhadijaMahanga) [inline wp-content/themes/hurumap/acf-json/group_6040a3626fa6e.json]: `Fontend` .. `Frontend`

File: wp-content/themes/hurumap/functions.php [hunk 3]
```diff
@@ -209,5 +209,42 @@ function bidirectional_acf_update_value( $value, $post_id, $field  ) {
...
+    $value = get_fields( 'options' );
+if (empty($value) ){
+    return $value;
+}else{
+    return "value";
+}
+    // $front_end__options = get_option( 'front_end__option_name' ); // Array of All Options
+    // return $front_end__options['base_url_0']; // Base Url
+}
+
```
Comment (KhadijaMahanga) [inline wp-content/themes/hurumap/functions.php]:
```suggestion
```

File: wp-content/themes/hurumap/functions.php [hunk 4]
```diff
@@ -209,5 +209,42 @@ function bidirectional_acf_update_value( $value, $post_id, $field  ) {
 function custom_index_name() {
     return 'outbreak';
 }
+function get_base_url() {
+    $value = get_fields( 'options' );
+if (empty($value) ){
+    return $value;
+}else{
+    return "value";
+}
```
Comment (KhadijaMahanga) [inline wp-content/themes/hurumap/functions.php]: Are we using this function anywhere? 

`get_fields(field, 'option')`

File: wp-content/themes/hurumap/functions.php [hunk 5]
```diff
@@ -209,5 +209,42 @@ function bidirectional_acf_update_value( $value, $post_id, $field  ) {
...
+    // return $front_end__options['base_url_0']; // Base Url
+}
+
 
 add_filter( 'ep_index_name', 'custom_index_name');
+
+
+//  Add custom preview page url link
+  function custom_preview_page_link($link) {
+    $base_url = get_field( 'description','options' );
```
Comment (KhadijaMahanga) [inline wp-content/themes/hurumap/functions.php]: I believe the field with the base_url is the `frontend_base_url` not `description`. And before proceeding, lets check if it is set.

File: wp-content/themes/hurumap/functions.php [hunk 6]
```diff
@@ -209,5 +209,42 @@ function bidirectional_acf_update_value( $value, $post_id, $field  ) {
...
+
+    $parentId = wp_get_post_parent_id( get_the_id());
```
Comment (KhadijaMahanga) [inline wp-content/themes/hurumap/functions.php:229]: `get_the_ID()` could return `false`

File: wp-content/themes/hurumap/functions.php (hunk 4 above)
Comment (KobbyMmo) [reply wp-content/themes/hurumap/functions.php]: @KhadijaMahanga  There is still a problem even when I do 
```
function get_base_url() {
    $value = get_field( 'frontend_base_url','options' );
    if ($value ){
        return $value;
    }else{
        return "";
    }
}
```

File: wp-content/themes/hurumap/functions.php (hunk 6 above)
Comment (KobbyMmo) [reply wp-content/themes/hurumap/functions.php:229]: @KhadijaMahanga  in what cases will it return false? and what is the foolproof way to check this?

**Review by KhadijaMahanga — APPROVED** [review]
🚀 🚀 

Just few comments. 
1. Not sure why we installed the jwt-auth plugin
2. There shouldn't be changes on `.htaccess`

**Comment by KobbyMmo** [comment]
> Not sure why we installed the jwt-auth plugin

@KhadijaMahanga We need it for API authentication

**Review by kilemensi — APPROVED** [review]
🚀 @KobbyMmo ... Works on my end (`JWT` is still giving me headaches though).

1. Lets make the Front-end URL setting the first one on the **HURUmap Site** option page.
Screenshot from 2021-03-09 13-06-31
2. To both you and @KhadijaMahanga : Is there any **_technical_** requirement for us to be bundling all the plugins we need in the plugins folder like this? Wouldn't simple solutions of just declaring plugins we depend on like [this](https://wordpress.stackexchange.com/a/131447) be better (if they work that is 🙂 ).

**Comment by KhadijaMahanga** [comment]
>  @KobbyMmo ... Works on my end (`JWT` is still giving me headaches though).
> 
> 1. Lets make the Front-end URL setting the first one on the **HURUmap Site** option page.
>    Screenshot from 2021-03-09 13-06-31
> 2. To both you and @KhadijaMahanga : Is there any **_technical_** requirement for us to be bundling all the plugins we need in the plugins folder like this? Wouldn't simple solutions of just declaring plugins we depend on like [this](https://wordpress.stackexchange.com/a/131447) be better (if they work that is slightly_smiling_face ).

@kilemensi 2. Yeah, I think declaring required plugin will be cool. We once looked into this, I am not sure what went wrong.

**Comment by KhadijaMahanga** [comment]
Oh, this is were the story ended 😆 https://github.com/TakwimuAfrica/Dashboard/pull/56 We can have a look at it again @kilemensi @KobbyMmo

**Comment by kilemensi** [comment]
> Oh, this is were the story ended laughing [TakwimuAfrica#56](https://github.com/TakwimuAfrica/Dashboard/pull/56) We can have a look at it again @kilemensi @KobbyMmo

Yo! In my mind we had already done that! 😢

**Pull request merged**